### ISSUE TITLE
Text inputs with prefixes or suffixes should follow our error message guidance

### ISSUE THREAD

**Issue by davidc-gds** [alphagov/govuk-design-system#2799]
This issue is from May 2023 external accessibility audit report.

**Usability issue**

**Issue ID:**
DAC_Usability_Error_Handling_01

**URLs:**
Example page: https://govuk-frontend-pr-3503.herokuapp.com/full-page-examples/make-a-payment#telephone-number

The related guidance section is '[Text inputs: error messages if the input has a prefix or suffix](https://design-system.service.gov.uk/components/text-input/#if-the-input-has-a-prefix-or-a-suffix)'

## Screen shots
<img width="808" alt="screenshot of the example text input with a £ prefix, showing that when the error summary pops up the input isn't receiving the proper red-outline error message styling." src="https://github.com/alphagov/govuk-frontend/assets/98318778/f64a58d7-a0b0-4721-b1d4-19acb62b3256">

After a user encounters an error on the ‘Make a payment’ page, the input field does not adopt an in-line error message, and the error skip link does not direct a user’s focus to the intended location.
The error skip link does not function because the ‘href’ attribute of the link, does not contain a value that matches the ID value of the <input> element it is intended to direct a user too.

## Current code ref(s)
`#main-content > form > div.govuk-error-summary > div > div > ul > li > a`

```
<a href="#telephone-number">Enter an amount, in pounds</a>
[…]
<input class="govuk-input govuk-input--width-5" id="payment-amount" name="payment-amount" type="text" spellcheck="false">
```

### Screen reader user
> “I intentionally committed an error to view the error handling. An error skip link appeared which was intended to move me directly to the error. However, when I selected the link, my focus remained on the link and did not move to the error as expected. I needed to manually move to the error which was more time consuming. Ensuring that focus moves to the intended target will allow me to correct the error easily and efficiently.”

### Keyboard only user
> “The error handling does not work as expected. When I activated the ‘enter an amount in pounds’ link with the enter key, nothing is activated, and my focus was not moved. I found this confusing and really think that I had either activated the wrong component of the page wasn’t working. Because of this I must move my focus manually with the tab key onto the edit field. This is much more time-consuming for me to do, and I did not expect this to happen. This also makes it inconsistent with previous error handling skip links. I would expect the skip link to activate when I press the enter key. I would expect every skip link to correlate with the component that needs correcting. So, when I press enter my focus should move directly onto corresponding component to correct it. This makes it faster and easier for me to understand what the issue is to be able to continue.”

### Voice activation user
>“When I activated the ‘enter amount in pounds’ link with label by name the link was highlighted however nothing was activated on the page. I found this extremely confusing and didn’t expect this to happen. This means I will have to use my focus into the edit field myself with the command click box. This is much more time-consuming to do and inconsistent with other skip links within the error handling. I would expect the skip link to automatically activate when I use label by name and move my focus into the corresponding component that need correcting this helps me to make the corrections quickly and easily with Dragon commands.”

## Solution
Error handling should follow the GOV.UK Design System guidelines with the relevant input field adopting an error message as well as the error skip links provided in the error summary, directing users to the intended location and the input field in error.
The link should possess a ‘href’ attribute with a value matching the ID value of input field in error.
Change the value of the ‘href’ to ‘#payment-amount’.

**Comment by querkmachine** [comment]
Was this just an issue in the modified pages we provided for testing? It would seem that we already advise on how to show error messages for this component in the Design System guidance. https://design-system.service.gov.uk/components/text-input/#if-the-input-has-a-prefix-or-a-suffix

**Comment by davidc-gds** [comment]
Hey @querkmachine yes that's correct, this one was specifically an issue with the prototype implementation.

@CharlotteDowns and I were thinking it might be a sign that we should strengthen our guidance if possible though. If we can get it wrong, that means other teams might be missing it and getting it wrong too.

I wouldn't say there's any code / dev / design work here. But there's potential for strengthening this specific existing guidance, either with a re-write or by bumping it up further on the page.

**Comment by CharlotteDowns** [comment]
I think I might close this as I think the guidance is strong enough I was just not good enough at cross-checking.
